lib/streamlit/type_util.py

```diff
@@ -241,9 +242,11 @@ def is_dataframe_like(obj: object) -> TypeGuard[DataFrameLike]:
 
 
 def is_snowpark_dataframe(obj: object) -> bool:
```

Comment:
nit: maybe it is better to rename this to be a bit more generic - e.g. something like `is_snowpark_data_object`.